### ISSUE TITLE
Sidenav toggle button doesn't clear the button focus state

### ISSUE THREAD

**Issue by mpoonuru** [angular/components#17749]
#### Reproduction

Use StackBlitz to reproduce your issue: https://stackblitz.com/fork/components-issue

Steps to reproduce:
1.  Click on sidenav toggle button  on the default Angular Material Example 
https://stackblitz.com/angular/rxplkxebyor?file=src%2Fapp%2Fsidenav-disable-close-example.ts

2. Close Reason backdrop clears the toggle button state but not the toggle button.
 
 
#### Expected Behavior

What behavior were you expecting to see?
Sidenav toggle button should work same as the backdrop click.

#### Actual Behavior

What behavior did you actually see?
Sidenav toggle button doesn't clear the cdk-focus on the nav button

#### Environment

- Angular: 8..2.15
- CDK/Material: 8.2.3
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Windows
![togglebutton-Angular](https://user-images.githubusercontent.com/24477255/69140542-b4c94080-0ae8-11ea-985e-dcc9e2e70e2f.png)
![backdrop-nav-angular](https://user-images.githubusercontent.com/24477255/69140588-c90d3d80-0ae8-11ea-8680-7189ee3ce1b5.png)

**Comment by crisbeto** [comment]
Focus is moved in both cases, we just set a different focus origin when restoring focus to the button which makes it look different. Buttons are styled so that they only show focus indication if focus came in from a keyboard interaction or programmatically.

**Comment by devversion** [comment]
Hello! It looks like clicking on the backdrop fails some `restoreFocus` checks, and the focus will not be properly restored. Hence clicking on the backdrop does not cause the button focus state.

It would actually be expected that also backdrop click shows the focus state on the button, plus it being actually focused.
